flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/call.scala

```diff
@@ -49,6 +57,128 @@ case class Call(functionName: String, args: Seq[Expression]) extends Expression
 }
 
 /**
+  * Over expression for calcite over transform.
+  *
+  * @param agg             over-agg expression
+  * @param aggAlias        agg alias for following `select()` clause.
+  * @param overWindowAlias over window alias
+  * @param overWindow      over window
+  */
+case class OverCall(
```

Comment:
Hm, not sure. Would need to look into this further. But, IMO it is very important to validate that the `partitionBy` fields exist.